tests/codemods/test_fix_mutable_params.py

```diff
@@ -191,5 +191,33 @@ def foo(x = None, y: Optional[List[int]] = None, z: Optional[Dict[str, int]] = N
     y = [] if y is None else y
     z = {} if z is None else z
     print(x, y, z)
+"""
+        self.run_and_assert(tmpdir, input_code, expected_output)
+
+    def test_fix_respect_docstring(self, tmpdir):
+        input_code = '''
+def func(foo=[]):
+    """Here is a docstring"""
+    print(foo)
+'''
+        expected_output = '''
+def func(foo=None):
+    """Here is a docstring"""
+    foo = [] if foo is None else foo
+    print(foo)
+'''
+        self.run_and_assert(tmpdir, input_code, expected_output)
+
+    def test_fix_respect_leading_comment(self, tmpdir):
+        input_code = """
+def func(foo=[]):
+    # Here is a comment
+    print(foo)
+"""
+        expected_output = """
+def func(foo=None):
+    foo = [] if foo is None else foo
+    # Here is a comment
```

Comment:
In general we can't know whether a comment was intended as belonging to the function decl or the first statement, so we don't preserve the position.